petastorm/unischema.py

```diff
@@ -221,7 +271,24 @@ def dict_to_spark_row(unischema, row_dict):
                 raise ValueError('Field {} is not "nullable", but got passes a None value')
         encoded_dict[field_name] = schema_field.codec.encode(schema_field, value) if value is not None else None
 
-    return Row(**encoded_dict)
+    if not unischema.column_bundles:
+        return Row(**encoded_dict)
+    else:
+        bundled_dict = {}
+        included_field_names = set()
+        for bundle_name, bundle_columns in unischema.column_bundles.items():
+            bundle_vals = [encoded_dict[field_name] for field_name in bundle_columns]
+            numpy_array = np.array(bundle_vals, dtype=object)
+            # Make a unischema field for the bundle to use the NdarrayCodec for serialization
+            unischema_field = UnischemaField(bundle_name, numpy_array.dtype, numpy_array.shape,
```

Comment:
Are you sure `NdarrayCodec` is a good level of reuse here? We don't use validation and 'faking' UnischemaField just to call the encode of the codec seems a bit awkward? If I understand correctly, you are really interested in these three lines:
```
        memfile = BytesIO()
        np.save(memfile, array)
        return bytearray(memfile.getvalue())
```
Maybe if Bundle becomes an separate object it can also serialize/deserialize itself?